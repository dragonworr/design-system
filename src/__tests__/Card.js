import React from 'react'
import renderer from 'react-test-renderer'
import { Card, theme } from '..'

describe('Card', () => {
  test('renders small box shadow with default border', () => {
    const json = renderer
      .create(<Card boxShadowSize="sm" theme={theme} />)
      .toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('box-shadow', '0 0 4px 0 rgba(0,0,0,0.08)')
    expect(json).toHaveStyleRule(
      'border',
      `1px solid ${theme.colors.borderGray}`
    )
  })

  test('renders medium box shadow with default border', () => {
    const json = renderer
      .create(<Card boxShadowSize="md" theme={theme} />)
      .toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule(
      'box-shadow',
      '0 0 4px 0 rgba(0,0,0,0.08),0 8px 8px 0 rgba(0,0,0,0.08)'
    )
    expect(json).toHaveStyleRule(
      'border',
      `1px solid ${theme.colors.borderGray}`
    )
  })

  test('renders large box shadow with default border', () => {
    const json = renderer
      .create(<Card boxShadowSize="lg" theme={theme} />)
      .toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule(
      'box-shadow',
      '0 0 4px 0 rgba(0,0,0,0.08),0 8px 8px 0 rgba(0,0,0,0.08),0 16px 16px 0 rgba(0,0,0,0.08)'
    )
    expect(json).toHaveStyleRule(
      'border',
      `1px solid ${theme.colors.borderGray}`
    )
  })

  test('renders xlarge box shadow with default border', () => {
    const json = renderer
      .create(<Card boxShadowSize="xl" theme={theme} />)
      .toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule(
      'box-shadow',
      '0 0 4px 0 rgba(0,0,0,0.08),0 8px 8px 0 rgba(0,0,0,0.08),0 16px 16px 0 rgba(0,0,0,0.08),0 32px 32px 0 rgba(0,0,0,0.08)'
    )
    expect(json).toHaveStyleRule(
      'border',
      `1px solid ${theme.colors.borderGray}`
    )
  })

  test('renders medium box shadow with specified borderWidth', () => {
    const json = renderer
      .create(<Card borderWidth={2} boxShadowSize="md" theme={theme} />)
      .toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule(
      'border',
      `2px solid ${theme.colors.borderGray}`
    )
    expect(json).toHaveStyleRule(
      'box-shadow',
      '0 0 4px 0 rgba(0,0,0,0.08),0 8px 8px 0 rgba(0,0,0,0.08)'
    )
  })
})
