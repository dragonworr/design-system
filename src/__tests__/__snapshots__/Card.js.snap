// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders large box shadow with default border 1`] = `
.c0 {
  box-shadow: 0 0 4px 0 rgba(0,0,0,0.08),0 8px 8px 0 rgba(0,0,0,0.08),0 16px 16px 0 rgba(0,0,0,0.08);
  border: 1px solid #d1d6db;
  border-radius: 2px;
}

<div
  className="c0 "
/>
`;

exports[`Card renders medium box shadow with default border 1`] = `
.c0 {
  box-shadow: 0 0 4px 0 rgba(0,0,0,0.08),0 8px 8px 0 rgba(0,0,0,0.08);
  border: 1px solid #d1d6db;
  border-radius: 2px;
}

<div
  className="c0 "
/>
`;

exports[`Card renders medium box shadow with specified borderRadius 1`] = `
.c0 {
  box-shadow: 0 0 4px 0 rgba(0,0,0,0.08),0 8px 8px 0 rgba(0,0,0,0.08);
  border: 1px solid #d1d6db;
  border-radius: 0px;
}

<div
  className="c0 "
/>
`;

exports[`Card renders medium box shadow with specified borderWidth 1`] = `
.c0 {
  box-shadow: 0 0 4px 0 rgba(0,0,0,0.08),0 8px 8px 0 rgba(0,0,0,0.08);
  border: 2px solid #d1d6db;
  border-radius: 2px;
}

<div
  className="c0 "
/>
`;

exports[`Card renders small box shadow with default border 1`] = `
.c0 {
  box-shadow: 0 0 4px 0 rgba(0,0,0,0.08);
  border: 1px solid #d1d6db;
  border-radius: 2px;
}

<div
  className="c0 "
/>
`;

exports[`Card renders xlarge box shadow with default border 1`] = `
.c0 {
  box-shadow: 0 0 4px 0 rgba(0,0,0,0.08),0 8px 8px 0 rgba(0,0,0,0.08),0 16px 16px 0 rgba(0,0,0,0.08),0 32px 32px 0 rgba(0,0,0,0.08);
  border: 1px solid #d1d6db;
  border-radius: 2px;
}

<div
  className="c0 "
/>
`;
